### ISSUE TITLE
[youtube] Cannot log in with TFA

### ISSUE THREAD

**Issue by ironiridis** [rg3/youtube-dl#10378]
### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2016.08.19*. If it's not read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2016.08.19**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones (n.b. [Proof that I searched](https://github.com/rg3/youtube-dl/issues/6365#issuecomment-240784684) )

### What is the purpose of your *issue*?
- [x] Bug report (encountered problems with youtube-dl)

```
$ ~/bin/youtube-dl -v -s "https://www.youtube.com/watch?v=MZUkG0OnSaA" 
[debug] System config: []
[debug] User config: ['-n', '--cookies', '~/.cache/youtube-dl.cookiejar', '--yes-playlist', '--console-title', '--add-metadata', '--mark-watched']
[debug] Command-line args: ['-v', '-s', 'https://www.youtube.com/watch?v=MZUkG0OnSaA']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.08.19
[debug] Python version 3.4.5 - Linux-4.7.0-vanilla+-x86_64-Intel-R-_Core-TM-2_Duo_CPU_P8600_@_2.40GHz-with-gentoo-2.2
[debug] exe versions: ffmpeg 2.8.7, ffprobe 2.8.7
[debug] Proxy map: {}
[youtube] Downloading login page
[youtube] Logging in
[youtube] Submitting TFA code
WARNING: Two-factor code expired or invalid. Please try again, or use a one-use backup code instead.
[youtube] MZUkG0OnSaA: Downloading webpage
[youtube] MZUkG0OnSaA: Downloading video info webpage
[youtube] MZUkG0OnSaA: Extracting video information
[youtube] MZUkG0OnSaA: Downloading MPD manifest
[youtube] MZUkG0OnSaA: Marking watched
```

I have captured the output of `--dump-pages`, compressed and uploaded it to my Google Drive: [--dump-pages output](https://drive.google.com/file/d/0B3kYvTu1uf-7QjB4dlhiNTlmVGs/view?usp=sharing)

**Comment by dstftw** [comment]
![image](https://cloud.githubusercontent.com/assets/1908898/17784329/4e4d1a06-65a6-11e6-9ea7-139f45564959.png)

**Comment by ironiridis** [comment]
Would it be possible for `youtube-dl` to indicate this somehow? I'm guessing this happened because I attempted to download a playlist and the TFA token was only valid for the first item.

**Comment by ironiridis** [comment]
I'd like to re-open this; I think I know what's going on.

I captured [a new --dump-pages](https://drive.google.com/file/d/0B3kYvTu1uf-7M0tmSFJTYnd2Tkk/view?usp=sharing) log. I went ahead and extracted the request responses and viewed them (honestly didn't think or realize that was a thing to do) and saw this:

![2016-08-19-144639_1280x800_scrot](https://cloud.githubusercontent.com/assets/308159/17822283/de00de10-661b-11e6-81bc-c4acc2edc8a3.png)

So unfortunately this seems we either need a way to specifically request code-based TFA, or detect this situation and abort. Critically, my phone *did not* prompt me to log in, so I'm wondering if Google is doing something here to try to detect automated login attempts... but that's another issue entirely.

I wonder if it would be feasible to have a `--launch-browser-for-login=/usr/bin/whatever` switch, where a user could specify they want `youtube-dl` to exec /usr/bin/whatever with the URL to complete login and obtain a token? Maybe to make this more generic and not require special-purpose hacks for each extractor.
